src/main/java/practice/StreamPractice.java

```diff
@@ -23,7 +37,11 @@ public int findMinEvenNumber(List<String> numbers) {
      * But before that subtract 1 from each element on an odd position (having the odd index).
      */
     public Double getOddNumsAverage(List<Integer> numbers) {
-        return 0D;
+        return IntStream
+                .range(0, numbers.size())
+                .map(i -> numbers.get(i) - i % 2)
+                .filter(i -> i % 2 != 0)
+                .average().getAsDouble();
```

Comment:
Do not combine operations in one row when others are single. It's not expected and can be missed during code reading
```suggestion
                .average()
                .getAsDouble();
```